src/UglyToad.PdfPig/DocumentLayoutAnalysis/RecursiveXYCut.cs

```diff
@@ -0,0 +1,408 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using UglyToad.PdfPig.Content;
+using UglyToad.PdfPig.Geometry;
+
+namespace UglyToad.PdfPig.DocumentLayoutAnalysis
+{
+    /// <summary>
+    /// The recursive X-Y cut is a top-down page segmentation technique that decomposes a document 
+    /// recursively into a set of rectangular blocks. This implementation leverages bounding boxes.
+    /// https://en.wikipedia.org/wiki/Recursive_X-Y_cut
+    /// <para>See 'Recursive X-Y Cut using Bounding Boxes of Connected Components' by Jaekyu Ha, Robert M.Haralick and Ihsin T. Phillips</para>
+    /// </summary>
+    public class RecursiveXYCut
+    {
+        /// <summary>
+        /// Get the blocks.
+        /// </summary>
+        /// <param name="pageWords">The words in the page.</param>
+        /// <param name="minimumWidth">The minimum width for a block.</param>
+        /// <param name="dominantFontWidth">The dominant font width.</param>
+        /// <param name="dominantFontHeight">The dominant font height.</param>
+        /// <returns></returns>
+        public static XYNode GetBlocks(IEnumerable<Word> pageWords, decimal minimumWidth,
```

Comment:
Is it worth an additional overload which takes just a `Page` and uses some preconfigured but overloadable defaults to help people fall into the pit of success?

e.g. 

```
public static XYNode GetBlocks(Page page, decimal? minimumWidth = null)
{
   return GetBlocks(page.GetWords(), minimumWidth.HasValue ? minimumWidth.Value : page.Width / 5, k => k.Max(), ...);
}```

Or something like that so that users can just 'call it' without too much knowledge and begin experimenting from there.

Reply:
yes, good idea but I think it's better to do the following:
`public static XYNode GetBlocks(IEnumerable<Word> pageWords, decimal minimumWidth = 0) { ... }`